packages/contracts-bedrock/test/properties/medusa/Protocol.properties.t.sol

```diff
@@ -87,7 +117,105 @@ contract ProtocolProperties is ProtocolHandler {
+        uint256 recipientIndex,
+        uint256 destinationChainId
+    )
+        external
+    {
+        property_BridgeSupertoken(fromIndex, recipientIndex, destinationChainId, 0);
+    }
+
+    /// @custom:property-id 9
+    /// @custom:property calls to relayERC20 with a value of zero dont modify accounting
+    /// @custom:property-id 7
+    /// @custom:property calls to relayERC20 always succeed as long as the cross-domain caller is valid
+    function property_RelayZeroDoesNotModifyAccounting(
+        uint256 fromIndex,
+        uint256 recipientIndex
+    )
+        external
+        withActor(msg.sender)
+    {
+        fromIndex = bound(fromIndex, 0, allSuperTokens.length - 1);
+        address recipient = getActorByRawIndex(recipientIndex);
+        OptimismSuperchainERC20 token = OptimismSuperchainERC20(allSuperTokens[fromIndex]);
+        uint256 balanceBefore = token.balanceOf(currentActor());
+        uint256 supplyBefore = token.totalSupply();
+
+        MESSENGER.setCrossDomainMessageSender(address(token));
+        vm.prank(address(MESSENGER));
+        try token.relayERC20(currentActor(), recipient, 0) {
+            MESSENGER.setCrossDomainMessageSender(address(0));
+        } catch {
+            // should not revert because of 7, and if it *does* revert, I want the test suite
+            // to discard the sequence instead of potentially getting another
+            // error due to the crossDomainMessageSender being manually set
+            assert(false);
+        }
+        uint256 balanceAfter = token.balanceOf(currentActor());
+        uint256 supplyAfter = token.totalSupply();
+        assert(balanceBefore == balanceAfter);
+        assert(supplyBefore == supplyAfter);
+    }
+
+    /// @custom:property-id 7
+    /// @custom:property calls to relayERC20 always succeed as long as the cross-domain caller is valid
+    /// @notice this ensures actors cant simply call relayERC20 and get tokens, no matter the system state
+    /// but there's still some possible work on how hard we can bork the system state with handlers calling
+    /// the L2ToL2CrossDomainMessenger or bridge directly (pending on non-atomic bridging)
+    function property_SupERC20RelayERC20AlwaysRevert(
+        uint256 tokenIndex,
+        address sender,
+        address crossDomainMessageSender,
+        address recipient,
+        uint256 amount
+    )
+        external
+    {
+        MESSENGER.setCrossDomainMessageSender(crossDomainMessageSender);
+        address token = allSuperTokens[bound(tokenIndex, 0, allSuperTokens.length)];
+        vm.prank(sender);
+        try OptimismSuperchainERC20(token).relayERC20(sender, recipient, amount) {
+            assert(sender == address(MESSENGER));
+            assert(crossDomainMessageSender == token);
+            // this would increase the supply across chains without a call to
+            // `mint` by the MESSENGER, so I'm reverting the state transition
+            require(false);
```

Comment:
here we are fully fuzzing the params to `relayERC20`. The fuzzer will, at some point, call the function with the correct params for it to not revert. That would increase the token total supply across chains without a call to `mint` by the bridge, which would break an invariant. However, an assumption of the campaign would also be broken in that case, which is that we'd have a `relayERC20` without a previous `sendERC20` in another chain, which is only valid because we mocked the parameters to it.

To not deal with that, I chose to (after ensuring I can explain _why_ the transaction succeded) revert the transaction, to not transition the state to one that can't be reached in production